src/components/Weather.jsx

```diff
@@ -0,0 +1,185 @@
+import Axios from 'axios';
+import React, { useState, useEffect } from 'react'
+import './Weather.css'
+
+
+
+
+function Weather() {
+  const API_key = "228b492fda91d5126216cb573939a770";
+
+  const [search, setSearch] = useState('');
+  const [img, setImg] = useState([]);
+  const [data, setData] = useState(null);
+  const [error, setError] = useState(false);
+
+  const bg = 'https://cdn.pixabay.com/photo/2020/02/06/15/59/forest-4824759_1280.png'
+
+  const fetchData = async () => {
+    try {
+      const response = await Axios.get(
+        `https://api.openweathermap.org/data/2.5/weather?q=${search}&appid=${API_key}&units=metric`
+      );
+      console.log(response.data);
+      setData(response.data);
+      setError(false);
+      
+    } catch (err) {
+      if (err.response && err.response.status === 404) {
+        // City not found error
+        setError(true);
+      } else {
+        // Other errors (e.g., rate limit exceeded)
+        console.error("Weather API Error:", err);
+        setError(true);
+      }
+      setData(null); // Reset data on error
+    }
+  };
+
+  const getImg = () => {
+    Axios.get(`https://api.unsplash.com/search/photos?page=1&query=${search}&client_id=vlWvWOTSt6UkqCFZ4gX76GqacFinpC_SMqmPahXo2RI`)
```

Comment:
Initialize variables for baseURL and cliant id and then merge it into single one
Eg. baseurl="www.wath...."
clientid:=abc
fullURL = baseURL+"search/photos?page=1&query="+ ${search}&client_id"=+clientid